server/py/services/alerts/crud/alerts.py

```diff
@@ -306,13 +308,16 @@ def _get_alert_state_cached(cls):
             )
         return cls._alert_state_cache
 
-    @staticmethod
-    def _try_populate_event_cache(session: sqlalchemy.orm.Session):
+    def _try_populate_caches(self, session: sqlalchemy.orm.Session):
         for alert in framework.utils.singletons.db.get_db().get_all_alerts(session):
+            # Populate events cache
             for event_name in alert.trigger.events:
                 services.alerts.crud.Events().add_event_configuration(
                     alert.project, event_name, alert.id
                 )
+            # Populate the alert and alert state caches
+            self._get_alert_by_id_cached()(session, alert.id)
+            self._get_alert_state_cached()(session, alert.id)
```

Comment:
Why do we need this and why on every iteration?

Reply:
My thought was that we anyway do this for loop for populating the events cache - so why not populating the other caches as well.
The initialization might be more costly, but later when endpoints are invoked we would already have the objects cached so no need to query the DB again.

Reply:
@TomerShor but it just initialises it initially once, which means that every iteration just returns same object every time, doesn't it?

Reply:
On the first iteration, the LRUCache itself is initialized with a registered callback.
When you call the cache with `(session, alert.id)`, it first searches for a relevant item in the cache. if not found - it triggers the callback.
The alert id is different in each iteration, so it populates the cache with the alert objects.